### ISSUE TITLE
Time warp fails on short spectrogram

### ISSUE THREAD

**Issue by tann9949** [zcaceres/spec_augment#12]
Hi, I have a question whether the time warping can sometimes cause a spectrogram to be too distorted or not. 

![Screen Shot 2563-07-22 at 14 37 29](https://user-images.githubusercontent.com/34812325/88148506-e0c53480-cc28-11ea-9572-b79dabd123e7.png)

I applied Time Warping with W=2 and it sometimes results in ugly padding on the right side of the spectrogram. Is there any way to alleviate this problem?

**Comment by shangeth** [comment]
Hey did you find what's wrong? I am using Time Warp with W=2(tried with W=1,...10)
I am getting different features at each time.

**run-1**
![Figure_1](https://user-images.githubusercontent.com/21142903/90485228-e2fdae80-e154-11ea-86aa-177f943bb1fc.png)

**run-2**
![Figure_2](https://user-images.githubusercontent.com/21142903/90485329-04f73100-e155-11ea-84fa-85be99458341.png)

**Comment by dtchantao** [comment]
> Hey did you find what's wrong? I am using Time Warp with W=2(tried with W=1,...10)
> I am getting different features at each time.
> 
> **run-1**
> ![Figure_1](https://user-images.githubusercontent.com/21142903/90485228-e2fdae80-e154-11ea-86aa-177f943bb1fc.png)
> 
> **run-2**
> ![Figure_2](https://user-images.githubusercontent.com/21142903/90485329-04f73100-e155-11ea-84fa-85be99458341.png)

same here, any solutions?

**Comment by PrathamSoni** [comment]
https://github.com/DemisEom/SpecAugment/blob/7f1435963b37ac8f9e4de9e44d754ecc41eaba85/SpecAugment/spec_augment_tensorflow.py#L46

It seems the implementation in this source code differs from the approach taken from in the above tensorflow implementation. This repo uses 2 points for fitting the transform, whereas the tf code uses multiple points

**Comment by zcaceres** [comment]
Great catch @PrathamSoni . Please feel free to PR. I'm afraid I have too much on my plate to fix this module.

**Comment by PrathamSoni** [comment]
@zcaceres Will open the PR within the next couple of days. 

This line in the bilinear interpolation is also deprecated (copying tensor from construct). I can patch that as well.
        alpha = torch.tensor((queries - floor), dtype=grid_type, device=grid_device)

**Comment by keshawnhsieh** [comment]
Got same situation. This BUG seriously affects the model performance. Any updates? @PrathamSoni

**Comment by PrathamSoni** [comment]
@keshawnhsieh just put up the pr. Could you let me know if it works better?

**Comment by keshawnhsieh** [comment]
> @keshawnhsieh just put up the pr. Could you let me know if it works better?

Many thanks. I will test it after Chinese New Year.

**Comment by zcaceres** [comment]
Fantastic @PrathamSoni . Thanks so much.

**Comment by PrathamSoni** [comment]
@thread especially @keshawnhsieh, I've made some fixes to the pull request. I think it should work perfectly now. LMK if you try it out!
